components/org.wso2.carbon.identity.conditional.auth.functions.http/src/test/java/org/wso2/carbon/identity/conditional/auth/functions/http/HTTPPostFunctionImplTest.java

```diff
@@ -491,4 +538,32 @@ public Map<String, String> dummyTokenEndpoint(@HeaderParam("Authorization") Stri
         }
         return response;
     }
+
+    @FunctionalInterface
+    public interface ResponseValidator {
```

Comment:
Seems like this interface and its impl are duplicated in each test classes. Let's write them outside the test classes separately.

Reply:
We can extract the Interface. Addressed with 46c8e19185f03a9c7a8e20a87f85cae4e5e0ea96